cloudinit/config/cc_rpi_interfaces.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2024, Raspberry Pi Ltd.
+#
+# Author: Paul Oberosler <paul.oberosler@raspberrypi.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit import subp
+from cloudinit.cloud import Cloud
+from cloudinit.config import Config
+from cloudinit.config.schema import MetaSchema
+from cloudinit.settings import PER_INSTANCE
+import logging
+
+
+LOG = logging.getLogger(__name__)
+RPI_INTERFACES_KEY = "rpi_interfaces"
+SUPPORTED_INTERFACES = {
+    "spi": "do_spi",
+    "i2c": "do_i2c",
+    "serial": "do_serial",
+    "onewire": "do_onewire",
+    "remote_gpio": "do_rgpio",
+    "ssh": "enable_ssh",
+}
+RASPI_CONFIG_SERIAL_CONS_FN = "do_serial_cons"
+RASPI_CONFIG_SERIAL_HW_FN = "do_serial_hw"
+
+meta: MetaSchema = {
+    "id": "cc_rpi_interfaces",
+    "distros": ["raspberry-pi-os"],
+    "frequency": PER_INSTANCE,
+    "activate_by_schema_keys": [RPI_INTERFACES_KEY],
+}
+
+
+# TODO: test
+def require_reboot(cfg: Config) -> None:
+    cfg["power_state"] = cfg.get("power_state", {})
+    cfg["power_state"]["mode"] = cfg["power_state"].get("mode", "reboot")
+    cfg["power_state"]["condition"] = True
+
+
+def is_pifive() -> bool:
+    try:
+        subp.subp(["/usr/bin/raspi-config", "nonint", "is_pifive"])
+        return True
+    except subp.ProcessExecutionError:
+        return False
+
+
+def configure_serial_interface(cfg: dict | bool, instCfg: Config) -> None:
+    enable_console = False
+    enable_hw = False
+
+    if isinstance(cfg, dict):
+        enable_console = cfg.get("console", False)
+        enable_hw = cfg.get("hardware", False)
+    elif isinstance(cfg, bool):
+        # default to enabling console as if < pi5
+        # this will also enable the hardware
+        enable_console = cfg
+
+    if not is_pifive() and enable_console:
+        # only pi5 has 2 usable UARTs
+        # on other models, enabling the console
+        # will also block the other UART
+        enable_hw = True
+
+    try:
+        subp.subp(
+            [
+                "/usr/bin/raspi-config",
+                "nonint",
+                RASPI_CONFIG_SERIAL_CONS_FN,
+                str(0 if enable_console else 1),
+            ]
+        )
+
+        try:
+            subp.subp(
+                [
+                    "/usr/bin/raspi-config",
+                    "nonint",
+                    RASPI_CONFIG_SERIAL_HW_FN,
+                    str(0 if enable_hw else 1),
+                ]
+            )
+        except subp.ProcessExecutionError as e:
+            LOG.error("Failed to configure serial hardware: %s", e)
+
+        require_reboot(instCfg)
+    except subp.ProcessExecutionError as e:
+        LOG.error("Failed to configure serial console: %s", e)
+
+
+def enable_ssh(cfg: Config, enable: bool) -> None:
```

Comment:
Do we need the separate tool to enable SSH? Cloud-init has built in SSH support. Is there a reason we can't use that? If so, is there a reason that module can't be expanded to do what this tool is doing?